src/Sentry.Unity.Android/SentryNativeAndroid.cs

```diff
@@ -35,9 +36,13 @@ public static void Configure(SentryUnityOptions options)
 
                     return crashedLastRun.Value;
                 };
-                // At this point Unity has taken the signal handler and will not invoke the original handler (Sentry)
-                // So we register our backend once more to make sure user-defined data is available in the crash report.
-                SentryNative.ReinstallBackend();
+
+                if (isIL2CPP)
```

Comment:
```suggestion
                // When running on Mono, we shouldn't take over the signal handler because its used to propagate exceptions into the VM.
                // If we take over, a C# null reference ends up crashing the app.
                if (isIL2CPP)
```